node/impl/full/eth.go

```diff
@@ -357,9 +354,17 @@ func (a *EthModule) EthGetTransactionByHashLimited(ctx context.Context, txHash *
 		return nil, nil
 	}
 
-	c, err := a.EthTxHashManager.GetCidFromHash(*txHash)
-	if err != nil {
-		log.Debug("could not find transaction hash %s in lookup table", txHash.String())
+	var c cid.Cid
+	var err error
+	if a.ChainIndexer != nil {
+		c, err = a.ChainIndexer.GetCidFromHash(ctx, *txHash)
+
+		if err != nil && errors.Is(err, index.ErrNotFound) {
+			log.Debug("could not find transaction hash %s in chain indexer", txHash.String())
```

Comment:
I'm trying to think through this error case - what does it mean for us to have an indexer, but to not find the txHash? Can we say with surety that in the case that we have an indexer then it should tell us about all the hashes we can find by other means and if it's not in there then it's not a valid hash we know about? maybe this is a `return nil, nil` case?

Reply:
@rvagg 

- We only index hashes for _ETH_ transactions.
- However, EVM explorers on Filecoin also index Filecoin messages in addition to ETH transactions. But, they use the ETH RPC API to fetch info about those Filecoin messages as well which means that they will pass the corresponding `eth_tx_hash` here.
- However, since that hash does not belong to an ETH TX, we wont find it in the ETH TX Index and the code below 

``if c == cid.Undef {
		c = txHash.ToCid()
	}
```

then converts it to a Filecoin msg cid to look up the corresponding state.

Reply:
ohhh, good point